src/imitation/util/networks.py

```diff
@@ -107,7 +107,11 @@ def update_stats(self, batch: th.Tensor) -> None:
     def forward(self, x: th.Tensor) -> th.Tensor:
         """Updates statistics if in training mode. Returns normalized `x`."""
         if self.training:
-            self.update_stats(x)
+            # Buffered tensors' gradients cannot be cleared by `optimizer.zero_grad()`.
+            # We need to avoid computing gradient for running_mean, running_var, and
+            # count, otherwise gradients get computed twice in consecutive iterations.
+            with th.no_grad():
```

Comment:
I looked into this more and I'm confident it's doing what we want now, thanks for catching this bug :)

Reply:
Oh, thank you for looking into this!